functions/__sf_section_rust.fish

```diff
@@ -0,0 +1,40 @@
+#
+# Rust
+#
+# Rust is a systems programming language sponsored by Mozilla Research.
+# Link: https://www.rust-lang.org
+
+function __sf_section_rust -d "Display the current Rust version"
+	# ------------------------------------------------------------------------------
+	# Configuration
+	# ------------------------------------------------------------------------------
+
+	__sf_util_set_default SPACEFISH_RUST_SHOW true
+	__sf_util_set_default SPACEFISH_RUST_PREFIX $SPACEFISH_PROMPT_DEFAULT_PREFIX
+	__sf_util_set_default SPACEFISH_RUST_SUFFIX $SPACEFISH_PROMPT_DEFAULT_SUFFIX
+	__sf_util_set_default SPACEFISH_RUST_SYMBOL "𝗥 "
+	__sf_util_set_default SPACEFISH_RUST_COLOR red
+
+	# ------------------------------------------------------------------------------
+	# Section
+	# ------------------------------------------------------------------------------
+
+	[ $SPACEFISH_RUST_SHOW = false ]; and return
+
+	if not test -f Cargo.toml \
+		-o (count *.rs) -gt 0
+		return
+	end
+
+	if not type -q rustc
+		return
+	end
+
+	set -l rust_version (rustc --version | grep --color=never -oE '[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]')
```

Comment:
This line is compatible with GNU grep (Most Linux systems), but not compatible with ripgrep, toybox's grep or busybox's grep. An if statement check, or using sed might be a good idea.

![image](https://user-images.githubusercontent.com/26250962/46438704-79d64300-c756-11e8-8bf6-a8301af2132c.png)

Here is a line that should do exactly the same thing: `rustc --version | sed 's/-.*//' | sed 's/.*\s//'`

Reply:
Good catch! 👍